azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -865,6 +893,64 @@ public Status getImpliedStatus(final ExecutableNode node) {
     return Status.READY;
   }
 
+  private Boolean isConditionMet(final ExecutableNode node) {
+    final String condition = node.getCondition();
+    if (condition == null) {
+      return true;
+    }
+
+    final Matcher matcher = VARIABLE_REPLACEMENT_PATTERN.matcher(condition);
+    String replaced = condition;
+
+    while (matcher.find()) {
+      final String value = findValueForJobVariable(matcher.group(1), matcher.group(2));
+      if (value != null) {
+        replaced = replaced.replace(matcher.group(), "'" + value + "'");
+      }
+      this.logger.info("Condition is " + replaced);
+    }
+
+    // Evaluate string expression using script engine
+    return evaluateExpression(replaced);
+  }
+
+  private String findValueForJobVariable(final String jobName, final String variable) {
+    // Get props from job props tmp file
+    final File jobPropsFile = findPropsFileForJob(jobName + "_props");
```

Comment:
Do we only have Prop_In and Prop_Out files given a job? Is jobName_Props_tmp is the naming template we currently follow. why is called "_tmp"? Is it possible that we can change its naming template?

Reply:
We only have <jobname>_output_***_ tmp and <jobname>_props_***_tmp files. See azkaban.jobExecutor.AbstractProcessJob#createFlattenedPropsFile and azkaban.jobExecutor.AbstractProcessJob#createOutputPropsFile for the naming. These are basically temp files. Why do you think we need to change the naming?

Reply:
Why the name followed up with `_tmp`?

Reply:
Because the files are created using File.createTempFile, it makes sense to me having _tmp in the name, e.g., jobA_output_1416309822424015744_tmp. This can also be easily distinguished from user's files in the execution directory.